server/src/main/java/com/objectcomputing/checkins/security/OAuthCsrfFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.objectcomputing.checkins.security;
+
+import io.micronaut.core.async.publisher.Publishers;
+import io.micronaut.http.*;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.cookie.Cookie;
+import io.micronaut.http.filter.OncePerRequestHttpServerFilter;
+import io.micronaut.http.filter.ServerFilterChain;
+import org.reactivestreams.Publisher;
+
+
+
+@Filter(value = {"/services/*", "/services/*"})
+public class OAuthCsrfFilter extends OncePerRequestHttpServerFilter {
+
+    @Override
+    protected Publisher<MutableHttpResponse<?>> doFilterOnce(HttpRequest<?> request, ServerFilterChain chain) {
+        HttpHeaders headers = request.getHeaders();
+
+        String headerValue = headers.get("X-CSRF-Header", String.class).orElse("");
+        String cookieValue = request.getCookies().findCookie("_csrf").map(Cookie::getValue).orElse(null);
+
+
+        if (!cookieValue.equals(headerValue)) {
```

Comment:
This seems to have broken all the unit tests.

Reply:
yaa, now I need to find a way to ignore this filter on tests otherwise I will need to modify tests. I think I will try adding @env=GCP and see if that take cares of it.

Reply:
If it's not intended to run locally then that would be a good fix. If it is you have the option of injecting this filter (or a mock of it) in the TestContainerSuite.

Reply:
I was able to add environment as GCP because it is not needed locally.